utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -169,21 +171,23 @@ impl<'a, 'b> SysCallsInvocator<'a, 'b> {
             self.unstructured.len()
         );
 
+        let mut invoke_instructions = vec![];
         for (invocable, (amount, call_indexes_handle)) in self.sys_call_imports.clone() {
-            let instructions =
-                self.build_sys_call_invoke_instructions(invocable, call_indexes_handle)?;
-
             log::trace!(
                 "Inserting the {} sys_call {} times",
                 invocable.to_str(),
                 amount
             );
 
-            for instructions in iter::repeat(&instructions).take(amount as usize) {
-                self.insert_sys_call_instructions(instructions)?;
+            for _ in 0..amount {
+                let instructions =
+                    self.build_sys_call_invoke_instructions(invocable, call_indexes_handle)?;
+                invoke_instructions.push(instructions);
```

Comment:
what were reasons for that? we will increase amount of "build sys-call" logs by "amount"

Reply:
I think it's more desirable to have different args to the syscalls, as they all have been inserted with the same args in the previous implementation

Reply:
wise, but make sure the logs will be well-organised and easy to investigate

Reply:
resolved?

Reply:
Are logs well organised after that and not confusing?